### PR TITLE
[feature](schema change) Light schema change support rollup/mv

### PR DESCRIPTION
**Pull request by swjtu-zhanglei**
# Proposed changes
1. move maxColUniqueId from OlapTable.java to MaterializedIndexMeta
2. for rollup and MaterializedIndex support light schema change
Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 1]
```diff
@@ -1851,4 +1840,8 @@ public void checkReplicaAllocation() throws UserException {
             }
         }
     }
+
+    public Boolean getUseLightSchemaChange() {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:1844]: Isn't it a property of OlapTable?
Also, naming it as `lightSchemaChangeEanbled()` seems better.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java [hunk 2]
```diff
@@ -1703,11 +1703,6 @@ private void createOlapTable(Database db, CreateTableStmt stmt) throws UserExcep
         OlapTable olapTable = new OlapTable(tableId, tableName, baseSchema, keysType, partitionInfo,
                 defaultDistributionInfo, indexes);
 
-        for (Column column : baseSchema) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java:1706]: We need to ensure that this change won't break the original behavior of base table.

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 3]
```diff
@@ -957,62 +937,97 @@ private boolean addColumnInternal(OlapTable olapTable, Column newColumn, ColumnP
...
-                checkAndAddColumn(modIndexSchema, newColumn, columnPos, newColNameSet, false);
+                IntSupplier targetIndexColUniqueIdSupplier = colUniqueIdSupplierMap.get(targetIndexId);
+                int rollUpNewColumnUniqueId = olapTable.getUseLightSchemaChange()
+                        ? Column.COLUMN_UNIQUE_ID_INIT_VALUE
+                        : targetIndexColUniqueIdSupplier.getAsInt();
+                checkAndAddColumn(modIndexSchema, newColumn, columnPos, newColNameSet,
+                        false, rollUpNewColumnUniqueId);
             }
         } else if (KeysType.DUP_KEYS == olapTable.getKeysType()) {
             if (targetIndexId == -1L) {
```
[the remark this answers is not in the document]
Comment (SWJTU-ZhangLei) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:969]: we cannot do like this, some situation we need both update base index and rollup index.

**Comment by gavinchou** [comment]
LGTM

**Comment by gavinchou** [comment]
LGTM again

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**